client/src/js/services/grid/GridGrouping.js

```diff
@@ -180,10 +180,30 @@ function GridGroupingService(uiGridGroupingConstants, $filter, Session, $timeout
     return groupingDetail.grouping[0].colName;
   }
 
-  /** return back the list of selected rows **/
+  /**
+   * return back the list of selected rows,
+   * if one row is selected so all transaction row will be considered as selected
+   * TO DO : make it generic, any kind of group should used, not only trans_id
+   **/
+  function getSelectedGroups (){
+
+    var parsed = [], processedTransactions = [];
```

Comment:
Just so that you know, every time you do
```js
function someFunction(values) {
  var array = [];
  values.forEach(function (row) {
    if (condition) {
      array.push(row);
    }
  });

  return array;
}
```
you are actually doing a [reduce](https://www.airpair.com/javascript/javascript-array-reduce).  Javascript has reduce [built in](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce).

I encourage you to get comfortable with reduce.  [Here is a really good course](http://reactivex.io/learnrx/) for mastering functional programming in Javascript.